rcl_action/test/rcl_action/test_action_interaction.cpp

```diff
@@ -0,0 +1,906 @@
+  test_msgs__action__Fibonacci_Result_Request incoming_result_request;
+  test_msgs__action__Fibonacci_Result_Response outgoing_result_response;
+  test_msgs__action__Fibonacci_Result_Response incoming_result_response;
+  test_msgs__action__Fibonacci_Feedback outgoing_feedback;
+  test_msgs__action__Fibonacci_Feedback incoming_feedback;
+  test_msgs__action__Fibonacci_Goal_Request__init(&outgoing_goal_request);
+  test_msgs__action__Fibonacci_Goal_Request__init(&incoming_goal_request);
+  test_msgs__action__Fibonacci_Goal_Response__init(&outgoing_goal_response);
+  test_msgs__action__Fibonacci_Goal_Response__init(&incoming_goal_response);
+  test_msgs__action__Fibonacci_Result_Request__init(&outgoing_result_request);
+  test_msgs__action__Fibonacci_Result_Request__init(&incoming_result_request);
+  test_msgs__action__Fibonacci_Result_Response__init(&outgoing_result_response);
+  test_msgs__action__Fibonacci_Result_Response__init(&incoming_result_response);
+  test_msgs__action__Fibonacci_Feedback__init(&outgoing_feedback);
+  test_msgs__action__Fibonacci_Feedback__init(&incoming_feedback);
+
+  // Initialize goal request
+  init_test_uuid0(outgoing_goal_request.uuid);
+  outgoing_goal_request.order = 10;
+
+  // Send goal request with valid arguments
+  int64_t sequence_number;
+  rcl_ret_t ret = rcl_action_send_goal_request(
+    &this->action_client, &outgoing_goal_request, &sequence_number);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_wait_set_clear(&this->wait_set);
+
+  ret = rcl_action_wait_set_add_action_server(&this->wait_set, &this->action_server, NULL);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Wait for goal request to be ready
+  ret = rcl_wait(&this->wait_set, RCL_S_TO_NS(1));
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  ret = rcl_action_server_wait_set_get_entities_ready(
+    &this->wait_set,
+    &this->action_server,
+    &this->is_goal_request_ready,
+    &this->is_cancel_request_ready,
+    &this->is_result_request_ready,
+    &this->is_goal_expired);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  EXPECT_TRUE(this->is_goal_request_ready) << rcl_get_error_string().str;
+  EXPECT_FALSE(this->is_cancel_request_ready) << rcl_get_error_string().str;
+  EXPECT_FALSE(this->is_result_request_ready) << rcl_get_error_string().str;
+
+  // Take goal request with valid arguments
+  rmw_request_id_t request_header;
+  ret = rcl_action_take_goal_request(&this->action_server, &request_header, &incoming_goal_request);
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  rcl_reset_error();
+
+  // Check that the goal request was received correctly
+  EXPECT_EQ(outgoing_goal_request.order, incoming_goal_request.order);
+  rcl_reset_error();
+  EXPECT_TRUE(uuidcmp(outgoing_goal_request.uuid, incoming_goal_request.uuid));
+  rcl_reset_error();
```

Comment:
This `rcl_reset_error()` is redundant with the previous since `uuidcmp` does not set an error. But again, we might be able to remove most, if not all, of the `rcl_reset_error()` calls.

Reply:
done, thanks!